lib/std/tz.zig

```diff
@@ -0,0 +1,218 @@
+            if (occur < 0) return error.Malformed; // rfc8536: occur [...] MUST be nonnegative
+            if (i > 0 and leapseconds[i - 1].occurrence + 2419199 > occur) return error.Malformed; // rfc8536: occur [...] each later value MUST be at least 2419199 greater than the previous value
+            if (occur > std.math.maxInt(i48)) return error.Malformed; // Unreasonably far into the future
+
+            const corr = std.mem.readIntSliceBig(i32, data2[p + 8 .. p + 12]);
+            if (i == 0 and corr != -1 and corr != 1) return error.Malformed; // rfc8536: The correction value in the first leap-second record, if present, MUST be either one (1) or minus one (-1)
+            if (i > 0 and leapseconds[i - 1].correction != corr + 1 and leapseconds[i - 1].correction != corr - 1) return error.Malformed; // rfc8536: The correction values in adjacent leap- second records MUST differ by exactly one (1)
+            if (corr > std.math.maxInt(i16)) return error.Malformed; // Unreasonably large correction
+
+            leapseconds[i] = .{
+                .occurrence = @intCast(i48, occur),
+                .correction = @intCast(i16, corr),
+            };
+            p += 12;
+        }
+
+        // Parse standard/wall indicators
+        i = 0;
+        while (i < isstdcnt) : (i += 1) {
+            const stdtime = data2[p];
+            if (stdtime == 1) {
+                timetypes[i].flags |= 0x02;
+            }
+            p += 1;
+        }
+
+        // Parse UT/local indicators
+        i = 0;
+        while (i < isutcnt) : (i += 1) {
+            const ut = data2[p];
+            if (ut == 1) {
+                timetypes[i].flags |= 0x04;
+                if (!timetypes[i].standardTimeIndicator()) return error.Malformed; // rfc8536: standard/wall value MUST be one (1) if the UT/local value is one (1)
+            }
+            p += 1;
+        }
+
+        // Footer
+        if (data2[p..].len < 2) return error.Malformed; // rfc8536 requires at least 2 newlines
+        if (data2[p] != '\n') return error.Malformed; // Not a rfc8536 footer
+        const footer_end = std.mem.indexOfScalar(u8, data2[p + 1 ..], '\n') orelse return error.Malformed; // No 2nd rfc8536 newline
+        const footer = try allocator.dupe(u8, data2[p + 1 .. p + 1 + footer_end]);
+        errdefer allocator.free(footer);
+
+        return Tz{
+            .allocator = allocator,
+            .transitions = transitions,
+            .timetypes = timetypes,
+            .leapseconds = leapseconds,
+            .footer = footer,
+        };
+    }
+
+    pub fn deinit(self: *Tz) void {
+        self.allocator.free(self.footer);
+        self.allocator.free(self.leapseconds);
+        self.allocator.free(self.transitions);
+        self.allocator.free(self.timetypes);
+    }
+};
+
+test "parse" {
+    // Asia/Tokyo is good for embedding, as Japan only had DST for a short while during the US occupation
+    const data = @embedFile("tz/asia_tokyo.tzif");
```

Comment:
Try and add a test file for each version. with and without legacy sections.

FWIW `America/Godthab` should be a small file.

Reply:
Done. Though Godthab isn't *that* small, so I added a fat version of Antarctica/Davis as a second test instead.